clippy_lints/src/semicolon_outside_block.rs

```diff
@@ -0,0 +1,160 @@
+use crate::rustc_lint::LintContext;
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::in_macro;
+use clippy_utils::source::snippet_with_macro_callsite;
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+use rustc_hir::intravisit::FnKind;
+use rustc_hir::ExprKind;
+use rustc_hir::{Block, Body, Expr, FnDecl, HirId, StmtKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::BytePos;
+use rustc_span::Span;
+use std::ops::Add;
+
+declare_clippy_lint! {
+    /// **What it does:** For () returning expressions, check that the semicolon is outside the block.
+    ///
+    /// **Why is this bad?** For consistency it's best to have the semicolon inside/outside the block. Either way is fine and this lint suggests outside the block.
+    /// Take a look at `semicolon_inside_block` for the other alternative.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// unsafe { f(x); }
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// unsafe { f(x) };
+    /// ```
+    pub SEMICOLON_OUTSIDE_BLOCK,
+    pedantic,
+    "add a semicolon outside the block"
+}
+
+declare_lint_pass!(SemicolonOutsideBlock => [SEMICOLON_OUTSIDE_BLOCK]);
+
+impl LateLintPass<'_> for SemicolonOutsideBlock {
+    fn check_fn(
```

Comment:
Why do we have to check a whole function here? Couldn't you just use `check_block` and then get the parent stmt and check if it is a `StmtKind::Semi`?